:book: Adding ComponentConfig Documentation (#1876)
kubernetes-sigs/kubebuilder: docs/book/src/component-config-tutorial/tutorial.md

# Tutorial: ComponentConfig

Nearly every project that is built for Kubernetes will eventually need to
support passing in additional configurations into the controller. These could
be to enable better logging, turn on/off specific feature gates, set the sync
period, or a myriad of other controls. Previously this was commonly done using
cli `flags` that your `main.go` would parse to make them accessible within your
program. While this _works_ it's not a future forward design and the Kubernetes
community has been migrating the core components away from this and toward

> using versioned config files, referred to as "componentconfig".

Reviewer: The phrase rather than the word is used throughout this tutorial
```suggestion
using versioned config files, referred to as "component configs".
```

The rest of this tutorial will show you how to configure your kubebuilder
project with the a component config type then moves on to implementing a custom
type so that you can extend this capability.

<aside class="note">

<h1>Following Along vs Jumping Ahead</h1>

Note that most of this tutorial is generated from literate Go files that

> form a runnable project, and live in the book source directory:
> [docs/book/src/componentconfig-tutorial/testdata/project][tutorial-source].

Reviewer: ```suggestion
form a runnable project, and live in this [source directory][tutorial-source].
```

Author: I was following the example from https://book.kubebuilder.io/multiversion-tutorial/tutorial.html still think I should change this?

Reviewer: In that case its probably ok to keep this as-is.
